code/WorkInProgress/MechanicsNetworkCon.dm

```diff
@@ -106,29 +109,11 @@
 		var/dataStr = ""//list2params(S.data)  Using list2params() will result in weird glitches if the data already contains a set of params, like in terminal comms
 		for(var/i in S.data)
 			dataStr += "[i][isnull(S.data[i]) ? ";" : "=[S.data[i]];"]"
-		var/datum/mechanicsMessage/msg = mechanics.newSignal(dataStr)
+		var/datum/mechanicsMessage/msg = mechanics.newSignal(dataStr, S.data_file?.copy_file())
 		mechanics.fireOutgoing(msg)
 		animate_flash_color_fill(src,"#00AA00",1, 1)
 		return
 
-	proc/post_file(var/target_id, var/key, var/value, var/file)
```

Comment:
You remove `post_file` here, but from what I can see, it's used by a bunch of procs in the codebase you haven't touched?

Reply:
I don't think this is used anywhere. There's an /obj/machinery/networked/proc/post_file, but the Mechcomp network connector isn't overriding that - it's a new proc that isn't actually called by anything that I can see.

Reply:
Ahh, a case of using the same name, got it.